### PR TITLE
Update dataset properties

### PR DESCRIPTION
**Pull request by matanper**
Add option to pass labels separated from data

### CODE REVIEW

File: deepchecks/base/dataset.py [hunk 1]
```diff
@@ -62,12 +62,13 @@ class Dataset:
     def __init__(
         self,
         df: pd.DataFrame,
+        label: pd.Series = None,
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:65]: why

File: makefile [hunk 2]
```diff
@@ -303,7 +303,7 @@ download:
 	$(PIP) install $(PROJECT)
 
 jupyter: $(JUPYTER)
-	$(BIN)/jupyter-notebook $(args) --notebook-dir=$(NOTEBOOK_CHECKS)
+	$(BIN)/jupyter-notebook $(args) --notebook-dir=./notebooks
```
Comment (ItayGabbay) [inline makefile:306]: why?

File: tests/base/dataset_test.py [hunk 3]
```diff
@@ -185,7 +185,7 @@ def test_dataset_infer_cat_features_max_categorical_ratio(diabetes_df):
 
 def test_dataset_label(iris):
     args = {'df': iris,
-            'label': 'target'}
+            'label_name': 'target'}
```
Comment (ItayGabbay) [inline tests/base/dataset_test.py:188]: What about test that tests the new label : pd.Series functionality?

**Review by ItayGabbay — CHANGES_REQUESTED** [review]
Why do we need these changes? 
Also - please add tests for the new added functionality

File: deepchecks/base/dataset.py (hunk 1 above)
Comment (matanper) [reply deepchecks/base/dataset.py:65]: I wrote in the PR description, to allow the labels to be passed separated from data. like we do with `from_numpy`

File: makefile (hunk 2 above)
Comment (matanper) [reply makefile:306]: because I want access to checks & examples, not only checks

File: tests/base/dataset_test.py (hunk 3 above)
Comment (matanper) [reply tests/base/dataset_test.py:188]: the `from_numpy` using it, so it tested implicitly

Comment (matanper) [reply tests/base/dataset_test.py:188]: fine, also added tests

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/base/dataset.py [hunk 4]
```diff
@@ -184,9 +192,9 @@ def from_numpy(
             *args: (np.ndarray):
                 expecting it to contain two numpy arrays (or at least one),
```
Comment (shir22) [inline deepchecks/base/dataset.py:193]: This phrasing is not clear (I know its not in the CR but good opportunity to improve it :) )

File: deepchecks/base/dataset.py [hunk 5]
```diff
@@ -184,9 +192,9 @@ def from_numpy(
...
                 first with features, second with labels.
-            feature_names (Sequence[Hashable], default None):
-                names for the feature columns. If not provided next names will
-                be assigned to the feature columns: X1-Xn (where n - number of features)
+            columns (Sequence[Hashable], default None):
+                names for the columns. If not provided next names will
+                be assigned to the feature columns: 1-n (where n - number of features)
```
Comment (shir22) [inline deepchecks/base/dataset.py:197]:
```suggestion
                names for the columns. If none provided, the names that will be automatically assigned
                to columns will be: 1-n (where n - number of columns)
```

Comment (shir22) [reply deepchecks/base/dataset.py:197]: in addition, why 1-n? I think that X_1 to X_n (or f_1 to f_n) is better. traditionally avoids confusion between column names and index names

File: deepchecks/base/dataset.py [hunk 6]
```diff
@@ -198,10 +206,9 @@ def from_numpy(
         Raises:
             DeepchecksValueError:
                 if receives zero or more than two numpy arrays;
-                if features (args[0]) is not two dimensional numpy array;
+                if columns (args[0]) is not two dimensional numpy array;
                 if labels (args[1]) is not one dimensional numpy array;
                 if features array or labels array is empty;
-                if features and labels arrays are not of the same size;
```
Comment (shir22) [inline deepchecks/base/dataset.py:204]: Why did you remove this line? (seems like an error...)

File: deepchecks/base/dataset.py [hunk 7]
```diff
@@ -198,10 +206,9 @@ def from_numpy(
         Raises:
             DeepchecksValueError:
                 if receives zero or more than two numpy arrays;
-                if features (args[0]) is not two dimensional numpy array;
+                if columns (args[0]) is not two dimensional numpy array;
                 if labels (args[1]) is not one dimensional numpy array;
                 if features array or labels array is empty;
```
Comment (shir22) [inline deepchecks/base/dataset.py:211]: Why is this an error? labels array can be None...
(and did you mean to rename to "features")

File: deepchecks/base/dataset.py [hunk 8]
```diff
@@ -184,9 +192,9 @@ def from_numpy(
...
                 expecting it to contain two numpy arrays (or at least one),
                 first with features, second with labels.
-            feature_names (Sequence[Hashable], default None):
-                names for the feature columns. If not provided next names will
-                be assigned to the feature columns: X1-Xn (where n - number of features)
+            columns (Sequence[Hashable], default None):
+                names for the columns. If not provided next names will
+                be assigned to the feature columns: 1-n (where n - number of features)
             label_name (Hashable, default None):
                 labels column name. If not provided next name will be used - 'target'
```
Comment (shir22) [inline deepchecks/base/dataset.py:199]:
```suggestion
                labels column name. If none is provided, the name 'target' will be used.
```

**Review by shir22 — DISMISSED** [review]

File: deepchecks/base/dataset.py [hunk 9]
```diff
@@ -78,19 +79,21 @@ def __init__(
...
-                List of names for the categorical features in the DataFrame. In order to disable categorical
+                List of names for the categorical features in the DataFrame. In order to disable categorical.
                 features inference, pass cat_features=[]
-            label (Optional[Hashable]):
+            label_name (Optional[Hashable]):
                 Name of the label column in the DataFrame.
             use_index (bool, default False):
                 Whether to use the dataframe index as the index column, for index related checks.
-            index (Optional[Hashable]):
+            index_name (Optional[Hashable]):
```
[the remark this answers is not in the document]
Comment (matanper) [reply deepchecks/base/dataset.py:94]: label and label_name can and should be used together. we create new dataframe with the label using the label_name.

File: deepchecks/base/dataset.py (hunk 5 above)
Comment (matanper) [reply deepchecks/base/dataset.py:197]: it's just the pandas default:
image

File: deepchecks/base/dataset.py (hunk 6 above)
Comment (matanper) [reply deepchecks/base/dataset.py:204]: I just moved this validation from method `from_numpy` to `init`, so left in the docstring only direct exceptions.

File: deepchecks/base/dataset.py (hunk 7 above)
Comment (matanper) [reply deepchecks/base/dataset.py:211]: it's not about features & labels, `arg[0]` is the columns array so it must be 2 dimensional (columns X samples)

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/base/dataset.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (shir22) [reply deepchecks/base/dataset.py:95]: Lets discuss this one orally, I have a question about this
(sounds possibly ambiguous to me)

File: deepchecks/base/dataset.py (hunk 7 above)
Comment (shir22) [reply deepchecks/base/dataset.py:211]: oh got it...
then only grammar here (is not *a* two/one dimensional numpy array)

File: tests/base/dataset_test.py [hunk 10]
```diff
@@ -544,7 +542,7 @@ def validate_dataset_created_from_numpy_arrays(
     label_column_name: str = 'target'
 ):
     if featue_columns_names is None:
-        featue_columns_names = [f'X{index}'for index in range(1, features_array.shape[1] + 1)]
+        featue_columns_names = [str(index) for index in range(1, features_array.shape[1] + 1)]
```
[the remark this answers is not in the document]
Comment (shir22) [reply tests/base/dataset_test.py]: lol, was referring to the typo in the name:
`featue_columns_names` (missing the "r")

File: deepchecks/base/dataset.py (hunk 4 above)
Comment (shir22) [reply deepchecks/base/dataset.py]: meant the sentence "expecting it to contain two numpy arrays (or at least one), first with features, second with labels."
What does it mean (or at least one), and what is the behavior for each scenario... (one/two)
